emperor/support_files/js/sceneplotview3d.js

```diff
@@ -100,6 +103,26 @@ define([
     this.dimensionRanges = {'max': [], 'min': []};
     this.drawAxesWithColor(0xFFFFFF);
     this.drawAxesLabelsWithColor(0xFFFFFF);
+
+    this.raycaster = new THREE.Raycaster();
+    this.mouse = new THREE.Vector2();
+
+    // Add callback call when sample is clicked
+    $container.on('click', function(event) {
```

Comment:
This is overall good, except that we can't easily test the raycasting/projection logic, do you think the bits inside this function could be abstracted so testing becomes more natural?